ui/src/criteria/concept.test.tsx

```diff
@@ -0,0 +1,154 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { EntityInstancesApiContext } from "apiContext";
+import { Cohort, Group, GroupKind } from "cohort";
+import { CohortUpdaterProvider } from "cohortUpdaterContext";
+import { ConceptCriteria } from "criteria/concept";
+import * as tanagra from "./tanagra-api";
+
+test.each([
+  [
+    "source",
+    {
+      instance: {
+        concept_name: {
+          stringVal: "test-concept",
+        },
+        concept_id: {
+          int64Val: 1234,
+        },
+        domain_id: {
+          int64Val: 5678,
+        },
+      },
+      matches: ["test-concept", "1234", "5678", "Source"],
+    },
+  ],
+  [
+    "standard",
+    {
+      instance: {
+        concept_name: {
+          stringVal: "test-concept",
+        },
+        concept_id: {
+          int64Val: 1234,
+        },
+        standard_concept: {
+          stringVal: "S",
+        },
+      },
+      matches: ["test-concept", "1234", "Standard"],
+    },
+  ],
+  [
+    "missing concept_id",
+    {
+      instance: {
+        concept_name: {
+          stringVal: "test-concept",
+        },
+      },
+      matches: ["0-0 of 0"],
+    },
+  ],
+])("%s", async (name, { instance, matches }) => {
+  await renderCriteria([instance]);
+
+  matches.forEach((match) => {
+    screen.getByText(match);
+  });
+});
+
+test("selection", async () => {
+  const { getCriteria, rerender } = await renderCriteria([
+    {
+      concept_id: {
+        int64Val: 100,
+      },
+    },
+    {
+      concept_id: {
+        int64Val: 101,
+      },
+    },
+  ]);
+
+  const checkboxes = screen.getAllByRole("checkbox");
+  expect(checkboxes.length).toBe(2);
+
+  expect(getCriteria().selected.length).toBe(0);
+
+  const getSelected = () => {
+    return getCriteria().selected.map((row) => row.id);
+  };
+
+  // Use waitFor because DataGrid has a focus update that occurs after the event
+  // (i.e. outside of act) which causes a warning.
+  await waitFor(async () => {
+    userEvent.click(checkboxes[0]);
```

Comment:
i don't think you want an event inside a waitFor.  i think if you had the linter enabled it would complain about this.

can you put the click first and then waitFor the expect?

Reply:
That's exactly what the comment is about. The click ends up calling setCellFocus sometime after the initial event which results in "Warning: An update to ForwardRef inside a test was not wrapped in act(...)." I've changed it to act() instead which also seems to be sufficient and doesn't require await but I don't otherwise know how to get rid of it.

Reply:
act(() => userEvent.click(checkboxes[0])); ?

I found userEvent to be a bit weird.  fireEvent never needs an act, it does that for you. userEvent supports some more complex interactions, and it seems like older versions needed to wrap in act, but newer versions didn't.  are you using the latest?  I don't think i needed the act in my code anymore (v13.5.0)

Reply:
Done.

It doesn't usually need the act but the data grid is doing something special that causes an extra update somehow and it isn't being handled automatically.